src/cpp/cpp_typecheck_constructor.cpp

```diff
@@ -105,10 +104,10 @@ static void copy_array(
   side_effect_exprt assign(ID_assign, typet(), source_location);
 
   assign.copy_to_operands(index_exprt(array.as_expr(), constant));
-  assign.op0().add_source_location() = source_location;
-
   assign.copy_to_operands(index_exprt(member, constant));
-  assign.op1().add_source_location() = source_location;
+
+  to_binary_expr(assign).op0().add_source_location() = source_location;
+  to_binary_expr(assign).op1().add_source_location() = source_location;
```

Comment:
As above: use the `side_effect_exprt` constructor that takes operands, and use some other approach instead of applying `to_binary_expr` on a `side_effect_exprt`.

Reply:
fixed as above